aviatrix/resource_aviatrix_aws_tgw_peering_domain_conn.go

```diff
@@ -20,27 +20,55 @@ func resourceAviatrixAWSTgwPeeringDomainConn() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"tgw_name1": {
-				Type:        schema.TypeString,
-				Required:    true,
-				ForceNew:    true,
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					tgwName2Old, _ := d.GetChange("tgw_name2")
+					domainName1Old, _ := d.GetChange("domain_name1")
+					domainName2Old, _ := d.GetChange("domain_name2")
+					return old == d.Get("tgw_name2").(string) && new == tgwName2Old.(string) &&
+						d.Get("domain_name1") == domainName2Old.(string) && d.Get("domain_name2") == domainName1Old.(string)
```

Comment:
Is the logic here correct?

Reply:
The logic here is that there should not be diff in this situation:

> new tgw1 == old tgw2
> new tgw2 == old tgw1
> new domain1 == old domain2
> new domain2 == old domain1

Reply:
This is too ugly. Please create a function for it and move it to aws_tgw_peering_domain_conn.go or wherever is appropriate.